### DISCUSSION TITLE
How to make "Inlay Hints"?

### DISCUSSION THREAD

**Discussion by ItsTheSky** [AvaloniaUI/AvaloniaEdit#429] [Q&A] [answered]
Hi there!
I'm looking toward re-creating an inlay hint system, but I'm not sure if it's possible, and if it is, where should I start?
Here's an example in Rider:
![image](https://github.com/AvaloniaUI/AvaloniaEdit/assets/50667548/042600b6-f773-4f69-8378-2006313a3b17)
Thanks!

**Comment by danipen** [comment, marked as answer]
It's something that is not supported at this moment. Only intra-column visuals are supported at this moment. Intra line visuals would need to be implemented.

There is also an issue here for tracking that. https://github.com/AvaloniaUI/AvaloniaEdit/issues/296

**Reply by ItsTheSky** [reply]
Got it, is there a way to add a sort of "hint" at the end of the line this time?
![image](https://github.com/AvaloniaUI/AvaloniaEdit/assets/50667548/5aca94ee-3eb8-4067-8a4a-027b1e880a28)

**Reply by danipen** [reply]
Yes I think you can achieve this using intra column adornments or using custom generators.

The demo project demonstrates how to add visuals (buttons) at caret position. I guess you can do that adding a text block at the end of the line.

**Comment by Gillibald** [comment]
I think this feature needs to be added to the VisualLine drawing visual
